tests/utils/test_requirements_utils.py

```diff
@@ -239,29 +238,29 @@ def test_get_installed_version(tmpdir):
 
     not_found_package = tmpdir.join("not_found.py")
     not_found_package.write("__version__ = '1.2.3'")
-    sys.path.insert(0, tmpdir.strpath)
+    monkeypatch.syspath_prepend(tmpdir.strpath)
     with pytest.raises(importlib_metadata.PackageNotFoundError, match=r".+"):
         importlib_metadata.version("not_found")
     assert _get_installed_version("not_found") == "1.2.3"
 
 
-def test_get_pinned_requirement(tmpdir):
+def test_get_pinned_requirement(tmpdir, monkeypatch):
     assert _get_pinned_requirement("mlflow") == f"mlflow=={mlflow.__version__}"
     assert _get_pinned_requirement("mlflow", version="1.2.3") == "mlflow==1.2.3"
 
     not_found_package = tmpdir.join("not_found.py")
     not_found_package.write("__version__ = '1.2.3'")
-    sys.path.insert(0, tmpdir.strpath)
+    monkeypatch.syspath_prepend(tmpdir.strpath)
     with pytest.raises(importlib_metadata.PackageNotFoundError, match=r".+"):
         importlib_metadata.version("not_found")
     assert _get_pinned_requirement("not_found") == "not_found==1.2.3"
 
 
-def test_get_pinned_requirement_local_version_label(tmpdir):
+def test_get_pinned_requirement_local_version_label(tmpdir, monkeypatch):
     package = tmpdir.join("my_package.py")
     lvl = "abc.def.ghi"  # Local version label
     package.write(f"__version__ = '1.2.3+{lvl}'")
-    sys.path.insert(0, tmpdir.strpath)
+    monkeypatch.syspath_prepend(tmpdir.strpath)
```

Comment:
Shall we use context like 
```
with monkeypatch.context() as m ` ?

Reply:
The second note in the page https://docs.pytest.org/en/4.6.x/monkeypatch.html use context. But other examples does not. I am not sure when we have to use context.

Reply:
In this case, we don't need to use `monkeypatch.context()`.

Reply:
https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.context says:

> Useful in situations where it is desired to undo some patches before the test ends, such as mocking stdlib functions that might break pytest itself if mocked (for examples of this see [issue #3290](https://github.com/pytest-dev/pytest/issues/3290)).


We don't have to undo pathces before the test ends.